image/manifest.go

```diff
@@ -59,54 +328,65 @@ func ReadSeedManifest(manifestFile string) (map[string]snap.Revision, error) {
 		}
 
 		tokens := strings.Fields(line)
-		// Expect exactly two tokens
-		if len(tokens) != 2 {
-			return nil, fmt.Errorf("line is illegally formatted: %q", line)
-		}
-
-		snapName := tokens[0]
-		revString := tokens[1]
-		if err := snap.ValidateName(snapName); err != nil {
-			return nil, err
-		}
 
-		rev, err := snap.ParseRevision(revString)
-		if err != nil {
-			return nil, err
+		if len(tokens) == 1 && strings.Contains(tokens[0], "/") {
```

Comment:
(super nitpick) I wonder if writing this as a flat switch/case is easier to read than the current nested if/then/else? Something like:
```go
		switch {
		case len(tokens) == 1 && strings.Contains(tokens[0], "/"):
			// Pinned validation-set: <account-id>/<name>=<sequence>
			if err := parsePinnedValidationSet(sm, tokens[0]); err != nil {
				return nil, err
			}
		case len(tokens) == 2 && strings.Contains(tokens[0], "/"):
			// Unpinned validation-set: <account-id>/<name> <sequence>
			if err := parseUnpinnedValidationSet(sm, tokens[0], tokens[1]); err != nil {
				return nil, err
			}
		case len(tokens) == 2:
			// Snap revision: <snap> <revision>
			if err := parseSnapRevision(sm, tokens[0], tokens[1]); err != nil {
				return nil, err
			}
		default:
			return nil, fmt.Errorf("cannot parse line: %q", line)
		}
	}
``` 
(maybe just preference but it feels slightly easier to read as one does not need to keep the state of the nesting in the head)

Reply:
Updated to this approach, I really like the switch statements and I'm trying to use them more. I forget they can be arbitrary in conditions